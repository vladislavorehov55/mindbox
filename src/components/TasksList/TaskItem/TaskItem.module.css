.task {
  display: flex;
  align-items: center;
  cursor: pointer;
  border-bottom: 1px solid rgb(118,118, 118);
  border-left: 1px solid rgb(118,118, 118);
  border-right: 1px solid rgb(118,118, 118);
  padding: 5px;
  height: 55px;
}
.completed {
  text-decoration: line-through;
  color: rgb(118, 118, 118);
}