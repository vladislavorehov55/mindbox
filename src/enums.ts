export enum TaskType {
  all = 'all',
  active = 'active',
  completed = 'completed'
}
export enum FilterType {
  all = 'all',
  active = 'active',
  completed = 'completed'
}