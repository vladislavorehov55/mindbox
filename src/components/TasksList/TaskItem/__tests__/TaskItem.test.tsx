import React from 'react';
import {mount, ReactWrapper} from 'enzyme';
import TaskItem from "../TaskItem";
import {TaskType} from "../../../../enums";

describe('TaskItem component', () => {
    let wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
    const completeTask = jest.fn()
    const getWrapper = () => mount(<TaskItem key={1} completeTask={completeTask} taskItem={{id: 1, text: 'Watch', type: TaskType.all}} />);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = getWrapper();
    });
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('should call completeTask function', () => {
        wrapper.find('li').simulate('click');
        expect(completeTask).toHaveBeenCalled();
    })
})