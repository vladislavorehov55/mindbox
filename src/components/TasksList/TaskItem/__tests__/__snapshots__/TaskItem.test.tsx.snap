// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem component should render correctly 1`] = `
<TaskItem
  completeTask={[MockFunction]}
  taskItem={
    {
      "id": 1,
      "text": "Watch",
      "type": "all",
    }
  }
>
  <li
    className="undefined false"
    onClick={[Function]}
  >
    <svg
      height="24px"
      version="1.1"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
      >
        <circle
          cx="12"
          cy="12"
          id="Circle"
          r="9"
          stroke="#6B6C6E"
          strokeWidth="1"
        />
      </g>
    </svg>
    Watch
  </li>
</TaskItem>
`;
